lesson-11/src/controllers/tasks.ts

```diff
@@ -0,0 +1,116 @@
+import { Request, Response } from 'express';
+import { z } from 'zod';
+import * as taskService from '../services/tasks';
+import { TaskFilters } from '../types/tasks';
+
+const createTaskSchema = z.object({
+  title: z.string().min(1),
+  description: z.string().optional(),
+  status: z.enum(['todo', 'in_progress', 'done']).optional(),
+  priority: z.enum(['low', 'medium', 'high']).optional()
+});
+
+const updateTaskSchema = createTaskSchema.partial();
+
+const querySchema = z.object({
+  createdAt: z.string().optional(),
+  status: z.string().optional(), // CSV or single
+  priority: z.string().optional() // CSV or single
+});
+
+function parseCsv(value?: string): string[] | undefined {
+  if (!value) return undefined;
+  return value.split(',').map(s => s.trim()).filter(Boolean);
+}
+
+export const getAllTasks = (req: Request, res: Response) => {
+  try {
+    const parseResult = querySchema.safeParse(req.query);
```

Comment:
валідація не має бути в контролері. контролер суто для того щоб приймати дані і викликати методи сервісу. для валідації краще створити окремі middleware функції

```suggestion
   router.get('/', validateQueryParams, getAllTasks);
```